### ISSUE TITLE
Allow authentication with Imgur

### ISSUE THREAD

**Issue by GameBurrow** [flameshot-org/flameshot#2740]
### Feature Description

At the end of the last year, I moved over from ShareX to Flameshot and have to say I love it.

One of the the features I'm missing quite a lot is to authenticate Flameshot with imgur so that it would actually upload the images as my account, allowing me to easily find them on the site and if needed even delete.

**Comment by borgmanJeremy** [comment]
No, that allows users to use thier own API key, but it is still an anonymous upload not associated with an account. We will need to push this to upload plugins.

**Comment by mmahmoudian** [comment]
@ZekXtreme 
> Any ETA on this ?

No. But if you want to work on it, Pull Requests are always welcome. 👐🏼

**Comment by alecmev** [comment]
Just in case, there's an abandoned (but, reportedly, working) PR here: https://github.com/flameshot-org/flameshot/pull/393

**Comment by sparr** [comment]
From that PR I found my way to this slightly updated fork, which is sadly still too old to cleanly merge into v12: https://github.com/jonpas/flameshot/tree/imgur-oauth2-v11.0.0

**Comment by sparr** [comment]
And here's my branch bringing that up to date with v12.1: https://github.com/sparr/flameshot/tree/imgur-oauth2-v12.1.0

Unfortunately this is a somewhat hack-y solution. It introduces a new top level config interface specifically for imgur, and also duplicates the client id config (the global one used for uploading with a custom client for api limit reasons, and a new one for authenticated uploads).

**Comment by navjotjsingh** [comment]
Any updates on this?
